clients/csharp-uwp/UWPVoiceAssistantSample/DirectLineSpeechDialogBackend.cs

```diff
@@ -108,75 +113,99 @@ public Task InitializeAsync(StorageFile keywordFile)
         {
             Contract.Requires(keywordFile != null);
 
-            // Default values -- these can be updated
-            this.ConnectorConfiguration = this.CreateConfiguration();
-            this.ConfirmationModel = KeywordRecognitionModel.FromFile(keywordFile.Path);
+            var configRefreshRequired = this.TryRefreshConfigValues();
 
-            this.connectorInputStream = AudioInputStream.CreatePushStream();
-            this.connector = new DialogServiceConnector(
-                this.ConnectorConfiguration,
-                AudioConfig.FromStreamInput(this.connectorInputStream));
+            var refreshConnector = configRefreshRequired || (this.keywordFilePath != keywordFile.Path);
 
-            this.connector.SessionStarted += (s, e) => this.SessionStarted?.Invoke(e.SessionId);
-            this.connector.SessionStopped += (s, e) => this.SessionStopped?.Invoke(e.SessionId);
-            this.connector.Recognizing += (s, e) =>
+            if (refreshConnector)
             {
-                switch (e.Result.Reason)
+                var newConnectorConfiguration = this.CreateConfiguration();
+
+                this.ConfirmationModel = KeywordRecognitionModel.FromFile(keywordFile.Path);
+                this.keywordFilePath = keywordFile.Path;
+                this.ConnectorConfiguration = newConnectorConfiguration;
+                this.connectorInputStream = AudioInputStream.CreatePushStream();
+
+                this.connector?.Dispose();
+                this.connector = new DialogServiceConnector(
+                    this.ConnectorConfiguration,

```

Comment:
Have you hit some issue like web socket maxi out maximum duration 300000 ms when you run app for a long time. Just curious as i hit its error in another app.

Reply:
Talked offline- the web socket times out, but the next call to a connector method reestablishes the connection. (I think the error I was seeing during the meeting was due to mic issues)